qa/workunits/windows/test_rbd_wnbd.py

```diff
@@ -101,6 +106,51 @@ def setup_logging(log_level: int = logging.INFO):
     LOG.setLevel(logging.DEBUG)
 
 
+def retry_decorator(timeout: int = 60,
+                    retry_interval: int = 2,
+                    silent_interval: int = 10,
+                    additional_details: str = "",
+                    retried_exceptions:
+                        typing.Union[
+                            typing.Type[Exception],
+                            collections.abc.Iterable[
+                                typing.Type[Exception]]] = Exception):
+    def wrapper(f: typing.Callable[..., typing.Any]):
+        @functools.wraps(f)
+        def inner(*args, **kwargs):
+            tstart: float = time.time()
+            elapsed: float = 0
+
+            while elapsed < timeout or not timeout:
+                try:
+                    return f(*args, **kwargs)
+                except retried_exceptions as ex:
+                    elapsed = time.time() - tstart
+                    details = (
+                        additional_details or "%s failed" % f.__qualname__)
+                    if timeout and elapsed > timeout:
+                        msg = (
+                            "Operation timed out. "
+                            "Additional details: %s. "
+                            "Exception: %s. "
+                            "Time elapsed: %d. Timeout: %d.")
+                        raise CephTestTimeout(
+                            msg % (details, ex, elapsed, timeout))
+
+                    if elapsed > silent_interval:
+                        level = logging.WARNING
+                    else:
+                        level = logging.DEBUG
+                    LOG.log(level,
+                            "Exception: %s. Additional details: %s. "
+                            "Time elapsed: %d. Timeout: %d",
+                            ex, details, elapsed, timeout)
+
+                    time.sleep(retry_interval)
```

Comment:
Should `elapsed` be updated after sleeping?  Otherwise `retry_interval` isn't (immediately) accounted for which not what the previous (inline) version did.